### ISSUE TITLE
Add Fish Shell Support

### ISSUE THREAD

**Issue by g5becks** [noplay/vscode-rainbow-end#29]
Hi, Thanks for the awesome plugin. I was wondering if it would be possible for you to add support for [fish shell](https://fishshell.com)

Here are a few code samples.

```
function _fisher_ls
    for pkg in $fisher_config/*/*/*
        command readlink $pkg; or echo $pkg
    end
end
```

```
switch $animal
    case cat
        echo evil
    case wolf dog human moose dolphin whale
        echo mammal
    case duck goose albatross
        echo bird
    case shark trout stingray
        echo fish
    case '*'
        echo I have no idea what a $animal is
end
```

```
if test -f foo.txt
    echo foo.txt exists
else if test -f bar.txt
    echo bar.txt exists
else
    echo foo.txt and bar.txt do not exist
end
```

```
while test -f foo.txt; or test -f bar.txt ; echo file exists; sleep 10; end
```

```
begin
    echo $xml_header
    echo $html_header
    if test -e $file
        ...
    end
    ...
end > out.html
```

Basically,  **_function_**,  _**begin**_ , _**for,**_ **_if_**, **_switch_**, and _**while**_ create block scopes that are closed witch the end keyword.

I did attempt to mess around with it for a second as seen below, but I actually not quite sure that I got it right.

```
  fish: {
      caseSensitive: true,
      ignoreInDelimiters:[
      {
        open: '"',
        close: '"'
      },
      {
        open: "'",
        close: "'"
      },
      {
        open: "--\\[\\[",
        close: "--\\]\\]"
      },
      {
        open: "--",
        close: "\n"
      },
      {open: "-", close: "\n"},
      
    ],
    inlineOpenTokens: [],
    openTokens: [
      "function",
      "if",
      "switch",
      "while",
      "begin",
      "for"
    ],
    closeTokens: ["end"],
    neutralTokens: [ "else", "case", "then", "in"]
  }
```

If this could be added, it would be greatly appreciated. 

Thanks.

**Comment by noplay** [comment]
Hello,

Are you using a specific extension for the coloration? Like this one https://marketplace.visualstudio.com/items?itemName=TeddyDD.fish ?

**Comment by g5becks** [comment]
Hi, yes - I am using this extension. https://marketplace.visualstudio.com/items?itemName=bmalehorn.vscode-fish

**Comment by noplay** [comment]
Hello I just released a 0.7.7 with support for it. Let me know if it's work fine

**Comment by g5becks** [comment]
@noplay looks good. 

Thanks.
